src/tools/rbd_mirror/ImageReplayer.cc

```diff
@@ -305,9 +305,17 @@ void ImageReplayer<I>::start(Context *on_finish, bool manual, bool restart)
       dout(10) << "canceled restart" << dendl;
       r = -ECANCELED;
     } else {
-      m_state = STATE_STARTING;
+      // checking if description is set to "orphan (force promote)"
+      // in which case we want to preserve the status.
+      if (m_state == STATE_STOPPED &&
+          m_state_desc.find("orphan") != std::string::npos) {
```

Comment:
Parsing error message/description like this is very fragile and never works out in the long run.  Also, it doesn't make sense to preserve _just_ `"orphan (force promote)"` description: we should either preserve all of them or not bother at all.

Reply:
Lets target the preserve part as part of a separate PR as discussed